### DISCUSSION TITLE
RAG embeds not visible in LibreChat UI

### DISCUSSION THREAD

**Discussion by mitchb95** [danny-avila/LibreChat#2572] [Troubleshooting] [answered]
Hello,

If I attach files through the LibreChat UI, I am able to use the embeddings and have conversations about their contents.  However, if I embed a file directly using the RAG API call, the file is not visible in the LibreChat UI.  Is this intended?  Are there any workarounds for this?

Thanks!

**Comment by Mixra** [comment, marked as answer]
The file metadata is not being stored in the main DB(info like file_id). A hacky way to do it would be to use /api/files from the backend API for LibreChat. Either way you're going to have to use the API from LibreChat since it has access to the main DB.
